api/app/controllers/concerns/spree/api/v2/storefront/order_concern.rb

```diff
@@ -26,7 +26,7 @@ def spree_current_order
           end
 
           def find_spree_current_order
-            Spree::Orders::FindCurrent.new.execute(
+            Spree::Api::Dependencies.storefront_current_order_finder.constantize.new.execute(
```

Comment:
Also, we could introduce some helper method that would call `constantize` only for non-constants
I guess it would be better to support both strings and constants as dependencies